test/sharness/t0700-remotepin.sh

```diff
@@ -319,6 +319,14 @@ test_remote_pins() {
 test_remote_pins ""
 
 test_kill_ipfs_daemon
+
+test_expect_success "'ipfs pin remote add' shows the warning message while offline" '
+  test_expect_code 0 ipfs pin remote add --service=test_pin_svc --enc=json $BASE_ARGS --name=name_a $HASH_A > actual &&
```

Comment:
I'm assuming here we are pinning something we _and also the service_ have so the command won't hang and still spit the warning which seems the best combination.

Reply:
It doesn't hang no matter it succeeds or fails.